src/qp_klp/Assays.py

```diff
@@ -144,6 +143,141 @@ def _generate_artifact_name(self, prep_file_path):
             # this is a normal pre-prep or sample-sheet.
             return (a_name, False)
 
+    def post_process_raw_fastq_output(self):
+        """This is an Amplicon specific method, see that object for
+           more info
+        """
+        pass
+
+    def quality_control(self):
+        """This is an MetaOmic specific method, see that object for
+           more info
+        """
+        pass
+
+    def execute_pipeline(self):
+        '''
+        Executes steps of pipeline in proper sequence.
+        :return: None
+        '''
+        if not self.is_restart:
+            self.pre_check()
+
+        # this is performed even in the event of a restart.
+        self.generate_special_map()
+
+        # even if a job is being skipped, it's being skipped because it was
+        # determined that it already completed successfully. Hence,
+        # increment the status because we are still iterating through them.
+
+        self.update_status("Converting data", 1, 9)
+        if "ConvertJob" not in self.skip_steps:
+            # converting raw data to fastq depends heavily on the instrument
+            # used to generate the run_directory. Hence this method is
+            # supplied by the instrument mixin.
+            self.convert_raw_to_fastq()
+            self.integrate_results()
+            self.generate_sequence_counts()
+
+        self.update_status("QC-ing reads", 2, 9)
+        if "NuQCJob" not in self.skip_steps:
+            # there is no failed samples reporting for amplicon runs.
+            self.post_process_raw_fastq_output()
+            self.quality_control()
+
+        self.update_status("Generating reports", 3, 9)
+        if "FastQCJob" not in self.skip_steps:
+            # reports are currently implemented by the assay mixin. This is
+            # only because metagenomic runs currently require a failed-samples
+            # report to be generated. This is not done for amplicon runs since
+            # demultiplexing occurs downstream of SPP.
+            self.generate_reports()
+
+        self.update_status("Generating preps", 4, 9)
+        if "GenPrepFileJob" not in self.skip_steps:
+            # preps are currently associated with array mixin, but only
+            # because there are currently some slight differences in how
+            # FastQCJob gets instantiated(). This could get moved into a
```

Comment:
A little confused by the mention of `FastQCJob` here since I thought we handled that at line 189 ... is this some sort of downstream consequence of that stuff?

Reply:
removing, left behind by mistake.